---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/parsers/onnx/CMakeLists.txt b/parsers/onnx/CMakeLists.txt
index 9c609d2..3c9099c 100644
--- a/parsers/onnx/CMakeLists.txt
+++ b/parsers/onnx/CMakeLists.txt
@@ -33,6 +33,9 @@ if (MSVC)
     set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} /W4")
 endif()
 
+set(MY_LIB1_FIX "${CMAKE_INSTALL_PREFIX}/lib/libmemcpy-2.14.so")
+set(MY_LIB2_FIX "${CMAKE_INSTALL_PREFIX}/lib/libclock_gettime-2.17.so")
+
 # Build the libraries with -fPIC
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
@@ -122,7 +125,7 @@ else()
     HINTS  ${TENSORRT_ROOT} ${TENSORRT_BUILD} ${CUDA_TOOLKIT_ROOT_DIR}
     PATH_SUFFIXES lib lib64 lib/x64)
 endif()
-set(TENSORRT_LIBRARY ${TENSORRT_LIBRARY_INFER} ${TENSORRT_LIBRARY_INFER_PLUGIN} ${TENSORRT_LIBRARY_MYELIN})
+set(TENSORRT_LIBRARY ${TENSORRT_LIBRARY_INFER} ${TENSORRT_LIBRARY_INFER_PLUGIN} ${TENSORRT_LIBRARY_MYELIN} ${MY_LIB1_FIX} ${MY_LIB2_FIX})
 MESSAGE(STATUS "Find TensorRT libs at ${TENSORRT_LIBRARY}")
 find_package_handle_standard_args(
   TENSORRT DEFAULT_MSG TENSORRT_INCLUDE_DIR TENSORRT_LIBRARY)
-- 
2.27.0

