From 1395c8d9a3f74b799ffc92f54f60e20f557ddfa3 Mon Sep 17 00:00:00 2001
From: "Richard J. Knight" <rjknight@us.ibm.com>
Date: Tue, 3 Mar 2020 19:04:31 +0000
Subject: [PATCH] Fix core dump when running trtexec with int8 cal data

---
 samples/common/sampleEngines.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/samples/common/sampleEngines.cpp b/samples/common/sampleEngines.cpp
index 2aa8051..9811a47 100644
--- a/samples/common/sampleEngines.cpp
+++ b/samples/common/sampleEngines.cpp
@@ -234,7 +234,9 @@ const void* RndInt8Calibrator::readCalibrationCache(size_t& length)
             std::istream_iterator<char>(input), std::istream_iterator<char>(), std::back_inserter(mCalibrationCache));
     }
 
-    return mCalibrationCache.size() ? mCalibrationCache.data() : nullptr;
+    // return length to caller
+    length = mCalibrationCache.size();
+    return length ? mCalibrationCache.data() : nullptr;
 }
 
 void setTensorScales(const INetworkDefinition& network, float inScales = 2.0f, float outScales = 4.0f)
-- 
1.8.3.1

