From 3a62a9252fd48af73072981f8c7ac8274ee63f0a Mon Sep 17 00:00:00 2001
From: "Richard J. Knight" <rjknight@us.ibm.com>
Date: Tue, 7 Jan 2020 17:57:28 +0000
Subject: [PATCH] fix tensorrt library version string

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b6b189c..4f1bb6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,8 +24,8 @@ set_ifndef(TRT_BIN_DIR ${CMAKE_BINARY_DIR})
 file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/include/NvInferVersion.h" VERSION_STRINGS REGEX "#define NV_TENSORRT_.*")
 
 foreach(TYPE MAJOR MINOR PATCH BUILD)
-    string(REGEX MATCH "NV_TENSORRT_${TYPE} [0-9]" TRT_TYPE_STRING ${VERSION_STRINGS})
-    string(REGEX MATCH "[0-9]" TRT_${TYPE} ${TRT_TYPE_STRING})
+    string(REGEX MATCH "NV_TENSORRT_${TYPE} [0-9]+" TRT_TYPE_STRING ${VERSION_STRINGS})
+    string(REGEX MATCH "[0-9]+" TRT_${TYPE} ${TRT_TYPE_STRING})
 endforeach(TYPE)
 
 foreach(TYPE MAJOR MINOR PATCH)
@@ -34,7 +34,7 @@ foreach(TYPE MAJOR MINOR PATCH)
 endforeach(TYPE)
 
 set(TRT_VERSION "${TRT_MAJOR}.${TRT_MINOR}.${TRT_PATCH}.${TRT_BUILD}" CACHE STRING "TRT project version")
-set(TRT_SOVERSION "${TRT_SO_MAJOR}.${TRT_SO_MINOR}.${TRT_SO_PATCH}" CACHE STRING "TRT library so version")
+set(TRT_SOVERSION "${TRT_SO_MAJOR}" CACHE STRING "TRT library so version")
 message("Building for TensorRT version: ${TRT_VERSION}, library version: ${TRT_SOVERSION}")
 
 if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
-- 
1.8.3.1

