From 071529907284cd05c31ab30c3e14bee478081167 Mon Sep 17 00:00:00 2001
From: Jason Furmanek <furmanek@us.ibm.com>
Date: Tue, 20 Jul 2021 15:17:30 -0400
Subject: [PATCH] add build onnx2trt flag

---
 CMakeLists.txt         | 3 ++-
 parsers/CMakeLists.txt | 2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a18cbd0..23d21ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,6 +56,7 @@ endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 option(BUILD_PLUGINS "Build TensorRT plugin" ON)
 option(BUILD_PARSERS "Build TensorRT parsers" ON)
 option(BUILD_SAMPLES "Build TensorRT samples" ON)
+option(BUILD_ONNX2TRT "Build onnx2trt executable" ON)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
@@ -184,7 +185,7 @@ else()
     find_library_create_target(nvinfer_plugin nvinfer_plugin SHARED ${TRT_OUT_DIR} ${TRT_LIB_DIR})
 endif()
 
-if(BUILD_PARSERS)
+if(BUILD_PARSERS OR BUILD_ONNX2TRT)
     add_subdirectory(parsers)
 else()
     find_library_create_target(nvcaffeparser nvparsers SHARED ${TRT_OUT_DIR} ${TRT_LIB_DIR})
diff --git a/parsers/CMakeLists.txt b/parsers/CMakeLists.txt
index a62afe4..4d5c647 100644
--- a/parsers/CMakeLists.txt
+++ b/parsers/CMakeLists.txt
@@ -14,6 +14,7 @@
 # limitations under the License.
 #
 
+if (BUILD_PARSERS)
 add_custom_target(parsers DEPENDS
     nvcaffeparserlibs
     nvonnxparser
@@ -31,4 +32,5 @@ include_directories(
    ${Protobuf_INCLUDE_DIR}
 )
 
+endif()
 add_subdirectory(onnx)
-- 
2.23.0

